src/test/java/seedu/logjob/logic/parser/ApplicationParserTest.java

```
@@ -0,0 +1,87 @@
+import static org.junit.jupiter.api.Assertions.assertThrows;
+import static org.junit.jupiter.api.Assertions.assertEquals;
+
+
+import seedu.logjob.logic.parser.exceptions.ParseException;
+import seedu.logjob.logic.commands.Command;
+import seedu.logjob.logic.commands.AddCommand;
+import seedu.logjob.logic.commands.DeleteCommand;
+import seedu.logjob.logic.commands.ExitCommand;
+import seedu.logjob.logic.commands.HelpCommand;
+import seedu.logjob.logic.commands.ListCommand;
+
+
+public class ApplicationParserTest {
+    private final ApplicationParser parser = new ApplicationParser();
+
+    @Test
+    public void parseCommand_validAddCommand_returnsAddCommand() throws Exception {
+        Command result = parser.parseCommand("add -n Google -j SWE -s 0");
+        Parser<AddCommand> expectedParser = new AddCommandParser();
+        Command expectedResult = expectedParser.parse(" -n Google -j SWE -s 0");
+
+        assertInstanceOf(AddCommand.class, result);
+        assertEquals(expectedResult, result);
+    }
+
+    @Test
+    public void parseCommand_validListCommand_returnsListCommand() throws Exception {
+        Command result = parser.parseCommand("list");
+        assertInstanceOf(ListCommand.class, result);
+    }
+
+    @Test
+    public void parseCommand_validHelpCommand_returnsHelpCommand() throws Exception {
+        Command result = parser.parseCommand("help");
+        assertInstanceOf(HelpCommand.class, result);
+    }
+
+    @Test
+    public void parseCommand_validDeleteCommand_returnsDeleteCommand() throws Exception {
+        Command result = parser.parseCommand("delete 1");
+        Parser<DeleteCommand> expectedParser = new DeleteCommandParser();
+        Command expectedResult = expectedParser.parse(" 1");
+
+        assertInstanceOf(DeleteCommand.class, result);
+        assertEquals(expectedResult, result);
+    }
+
+    @Test
+    public void parseCommand_validExitCommand_returnsExitCommand() throws Exception {

```

Could use more negative testing for single word commands. For instance `exit -1234` should throw an exception.